File: Sorting/InsertionSort/insertionSort.cpp
#include<iostream>
using namespace std;

void printArray(int arr[], int n)
{
    for(int i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }
}

void insertionSort(int arr[], int n)
{
    for(int i = 1; i < n; i++)
    {
        int key = arr[i];
        int j = i - 1;
        while(key < arr[j] && j >= 0)
        {
            arr[j + 1] = arr[j];
            --j;
        }
        arr[j + 1] = key;
    }
}

int main()
{
    int data[] = {43, 2, 54, -54, 232, -5432, 0};
    int size = sizeof(data) / sizeof(data[0]);
    insertionSort(data, size);
    printArray(data, size);

    return 0;
}

File: Sorting/InsertionSort/insertionSort.py
def insertionSort(arr,n):
    for i in range(1,n):
        key = arr[i]
        j = i-1
        while(key < arr[j] and j>=0):
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = key
        
def printArray(arr,n):
    for i in range(n):
        print(arr[i])
        
data = [3,34,1,-34,222,0,43,-2]
insertionSort(data,len(data))
print(data,len(data))